### ISSUE TITLE
Microceph does not come with an orchestrator backend module

### ISSUE THREAD

**Issue by FSeidinger-XI** [canonical/microceph#434]
# Issue report

There is currently no orchestrator backend deployed/packaged in the microceph snap distribution

## What version of MicroCeph are you using ?

reef/stable: 18.2.4+snapc9f2b08f92

## What are the steps to reproduce this issue ?

1. Open the dashboard
2. Change to menu "Cluster / Physical Disks"
3. See the warning from the dashboard
   "Orchestrator is not available
   Orchestrator is unavailable: Module not found
   Please consult the [documentation](https://docs.ceph.com/en/latest/mgr/orchestrator) on how to configure and enable the Orchestrator management functionality."
4. Read the manual and try to set the orchestrator backend with:
   `ceph orch set backend cephadm` or `ceph orch set backend rook`
5. List the modules with `ceph mgr module ls`

## What happens (observed behaviour) ?

For both modules `cephadm` and `rook` you get the error `Error EINVAL: Module 'cephadm' not found` or `Error EINVAL: Module 'rook' not found.

The list of manager modules reveals that there are no cephadm or rook modules present.

```shell
MODULE
balancer           on (always on)
crash              on (always on)
devicehealth       on (always on)
orchestrator       on (always on)
pg_autoscaler      on (always on)
progress           on (always on)
rbd_support        on (always on)
status             on (always on)
telemetry          on (always on)
volumes            on (always on)
dashboard          on
insights           on
iostat             on
nfs                on
osd_perf_query     on
osd_support        on
prometheus         on
restful            on
alerts             -
influx             -
localpool          -
mirroring          -
selftest           -
snap_schedule      -
stats              -
telegraf           -
test_orchestrator  -
zabbix             -
```
## What were you expecting to happen ?

There is at least one suitable orchestrator backend coming with the microceph distribution in order to use the full potential of the dashboard.

**Comment by UtkarshBhatthere** [comment]
@FSeidinger-XI unfortunately, MicroCeph does not work like Cephadm and Rook, hence, you cannot use the `cephadm` or `rook` backend with `microceph`. Ideally, this needs a new orchestrator module in ceph which can talk to MicroCeph and trigger ceph orchestration. For the time being, please refer to our [docs](https://canonical-microceph.readthedocs-hosted.com/en/latest/how-to/)...... `sudo microceph disk` command should expose all the required commands to work with OSDs. :D

**Comment by FSeidinger-XI** [comment]
> @FSeidinger-XI unfortunately, MicroCeph does not work like Cephadm and Rook, hence, you cannot use the `cephadm` or `rook` backend with `microceph`. Ideally, this needs a new orchestrator module in ceph which can talk to MicroCeph and trigger ceph orchestration. For the time being, please refer to our [docs](https://canonical-microceph.readthedocs-hosted.com/en/latest/how-to/)...... `sudo microceph disk` command should expose all the required commands to work with OSDs. :D

Yeah, I guess I already guessed about that. But its not only about the disks. Some other stuff that can be done from the GUI, like administering service or bringing up and configure the Object gateway will also not work.

Since Microceph is about simplicity and user experience, I would say that a functioning GUI is a must have.

**Comment by FSeidinger-XI** [comment]
To get this topic running. Is there any interface specification about how the frontend is communicating with the backend and how the backend have to implement this interfaces.

Maybe then someone can take up this task and implement it? I'm no expert in go, but I'm pretty good in programming and architecture. Besides my professional work, I may spend some thoughts and work on this.

**Comment by sabaini** [comment]
I think we should clarify the docs wrt to orchestrator / dashboard support

**Comment by abasu0713** [comment]
> > @FSeidinger-XI unfortunately, MicroCeph does not work like Cephadm and Rook, hence, you cannot use the `cephadm` or `rook` backend with `microceph`. Ideally, this needs a new orchestrator module in ceph which can talk to MicroCeph and trigger ceph orchestration. For the time being, please refer to our [docs](https://canonical-microceph.readthedocs-hosted.com/en/latest/how-to/)...... `sudo microceph disk` command should expose all the required commands to work with OSDs. :D
> 
> Yeah, I guess I already feared about that. But its not only about the disks. Some other stuff that can be done from the GUI, like administering service or bringing up and configure the Object gateway will also not work.
> 
> Since Microceph is about simplicity and user experience, I would say that a functioning GUI is a must have.

I don't fully agree. Microceph is designed for all of production/edge/local scenarios. You don't necessarily need a GUI for configuring Ceph Cluster..  I have seen a Rook-Ceph operator connected to an external cluster works just fine along with Object Gateways.. Additionally Microceph has simplified Rados/Object GW to a bare minimum command. You can now spin a [globally accessible S3 Compatible Object Store](https://medium.com/@arko.basu09/part-1-diy-globally-accessible-s3-compatible-object-store-1a625460e802) in less than 5 min with full S3 Compatibility which is faster than on any cloud provider in my opinion. 
Pardon my candor.
